File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
# 프로그래머스 43165 타겟 넘버 LV.2
answer = 0
def dfs(idx, numbers, target, value):
    global answer
    l = len(numbers)
    if idx == l:
        if target == value:
            answer += 1
        return

    dfs(idx + 1, numbers, target, value + numbers[idx])
    dfs(idx + 1, numbers, target, value - numbers[idx])


def solution(numbers, target):
    dfs(0, numbers, target, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    
    public void findCount(int[] numbers, int target, int idx, int value) {
        if (idx == numbers.length) {
            if (value == target) {
                this.answer++;
            }
            return;
        }
        
        findCount(numbers, target, idx + 1, value + numbers[idx]);
        findCount(numbers, target, idx + 1, value - numbers[idx]);
    }
    
    
    public int solution(int[] numbers, int target) {
        findCount(numbers, target, 0, 0);
        return answer;
    }
}